src/material/core/tokens/m2/mdc/_button-text.scss

```diff
@@ -0,0 +1,112 @@
+@use 'sass:map';
+@use '../../../theming/theming';
+@use '../../../typography/typography-utils';
+@use '../../token-utils';
+
+// The prefix used to generate the fully qualified name for tokens in this file.
+$prefix: (mdc, text-button);
+
+// Tokens that can't be configured through Angular Material's current theming API,
+// but may be in a future version of the theming API.
+//
+// Tokens that are available in MDC, but not used in Angular Material should be mapped to `null`.
+// `null` indicates that we are intentionally choosing not to emit a slot or value for the token in
+// our CSS.
+@function get-unthemable-tokens() {
+  @return (
+    // The border radius of text buttons.
+    container-shape: 4px,
+    // The text transform used for text buttons.
+    label-text-transform: none,
+    // The size of icons used inside text buttons.
+    with-icon-icon-size: 1.125rem,
+    // =============================================================================================
+    // = TOKENS NOT USED IN ANGULAR MATERIAL                                                       =
+    // =============================================================================================
+    // This is an odd token that does not result in a runtime configurable CSS variable. Instead it
+    // toggles emission of some styles in the sass mixin. We don't use this in Angular.
+    keep-touch-target: null,
+  );
+}
+
+// Tokens that can be configured through Angular Material's color theming API.
+@function get-color-tokens($config) {
+  $foreground: map.get($config, foreground);
+  $is-dark: map.get($config, is-dark);
+
+  @return (
+    // The text color for text buttons in the disabled state.
+    disabled-label-text-color: theming.get-color-from-palette($foreground, disabled-text),
+    // The text color for text buttons in the focused state.
+    focus-label-text-color: theming.get-color-from-palette($foreground, base),
```

Comment:
It seems like `theming.get-color-from-palette($foreground, base)` is repeated a lot in this function. Maybe it should be moved into a local variable?